src/buttons/CopyToClipboard/index.jsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'preact/hooks'
+import PropTypes from 'prop-types'
+import CSS from './styles.css'
+
+import copyToClipboard from './../../utils/copy-to-clipboard'
+
+const CopyToClipboard = ({
+	dataTestId,
+	valueToCopy,
+	labelToCopy,
+	labelCopying,
+	labelCopied,
+	classToCopy,
+	classCopying,
+	classCopied,
+	copyingFeedbackTime,
+	copiedFeedbackTime,
+}) => {
+	const TO_COPY = { class: classToCopy, label: labelToCopy }
+	const COPYING = { class: classCopying, label: labelCopying }
+	const COPIED = { class: classCopied, label: labelCopied }
+
+	const COPY_TIMEOUT = copyingFeedbackTime
+	const COPIED_TIMEOUT = copiedFeedbackTime
+
+	const [copyStatus, setCopyStatus] = useState(TO_COPY)
+
+	const afterCopy = () => {
+		setCopyStatus(COPIED)
+		setTimeout(() => {
+			setCopyStatus(TO_COPY)
+		}, COPIED_TIMEOUT)
+	}
+
+	const handleOnClick = textToCopy => {
+		setCopyStatus(COPYING)
+		setTimeout(() => {
+			copyToClipboard(textToCopy, afterCopy)
```

Comment:
(minor) it seems that this `copyToClipboard` function accepts a callback as its second argument. This would also be a good candidate for refactoring into an `async` function, that way you could do:

```js
await copyToClipboard(textToCopy);
afterCopy();
```

Reply:
Please feel free to add a task for changing it later unless it's reasonable to change as part of this PR.